core/sql/optimizer/RelGrby.h

```diff
@@ -152,6 +154,9 @@ class GroupByAgg : public RelExpr
   inline void setGroupExpr(ValueIdSet &expr) { groupExpr_ = expr;}
   inline void addGroupExpr(ValueIdSet &expr) { groupExpr_ += expr;}
 
+  inline ValueIdList & extraOrderExpr() { return extraOrderExpr_; }
```

Comment:
Returning a writable reference to a data member isn't really a good idea. I know it's done in many other places, but it's still not good. Do you really need this?